lib/js/src/manager/screen/menu/_MenuManagerBase.js

```diff
@@ -0,0 +1,447 @@
+        this._cancelAllTasks(); // cancel all operations
+        // remove the listeners
+        if (typeof this._hmiListener === 'function') {
+            this._lifecycleManager.removeRpcListener(FunctionID.OnHMIStatus, this._hmiListener);
+        }
+        if (typeof this._commandListener === 'function') {
+            this._lifecycleManager.removeRpcListener(FunctionID.OnCommand, this._commandListener);
+        }
+        if (typeof this._onDisplayCapabilityListener === 'function') {
+            this._lifecycleManager.removeOnSystemCapabilityListener(SystemCapabilityType.DISPLAYS, this._onDisplayCapabilityListener);
+        }
+        super.dispose();
+    }
+
+    /**
+     * Suspend the queue if the HMI level is NONE since we want to delay sending RPCs until we're in non-NONE
+     */
+    _updateTransactionQueueSuspended () {
+        if (this._currentHmiLevel === HMILevel.HMI_NONE || this._currentSystemContext === SystemContext.SYSCTXT_MENU) {
+            console.log(`MenuManagerBase - Suspending the transaction queue. Current HMI level is: ${this._currentHmiLevel}, current system context is: ${this._currentSystemContext}`);
+            this._canRunTasks = false;
+        } else {
+            console.log('MenuManagerBase - Starting the transaction queue');
+            this._canRunTasks = true;
+            this._invokeTaskQueue();
+        }
+    }
+
+    /**
+     * Sets the dynamic updates mode
+     * @param {DynamicMenuUpdatesMode} value - The dynamic updates mode
+     */
+    _setDynamicMenuUpdatesMode (value) {
+        this._dynamicMenuUpdatesMode = value;
+    }
+
+    /**
+     * Gets the dynamic updates mode
+     * @returns {DynamicMenuUpdatesMode} - The dynamic updates mode
+     */
+    _getDynamicMenuUpdatesMode () {
+        return this._dynamicMenuUpdatesMode;
+    }
+
+
+    /**
+     * Creates and sends all associated Menu RPCs
+     * @param {MenuCell[]} cells - The menu cells that are to be sent to the head unit, including their sub-cells.
+     */
+    _setMenuCells (cells = []) {
+        if (cells === null) {
+            console.error('MenuManagerBase - Cells list is null. Skipping...');
+            return;
+        }
+
+        if (!this._menuCellsAreUnique(cells.map(cell => cell.clone()), [])) {
+            console.error('MenuManagerBase - Not all set menu cells are unique, but that is required');
+            return;
+        }
+
+        // Create a deep copy of the list so future changes by developers don't affect the algorithm logic
+        this._menuCells = cells.map(cell => cell.clone());
+        const isDynamicMenuUpdateActiveBoolean = this._isDynamicMenuUpdateActive(this._dynamicMenuUpdatesMode, this._displayType);
+
+        const operation = new _MenuReplaceOperation(this._lifecycleManager, this._fileManager, this._windowCapability,
+            this._menuConfiguration, this._currentMenuCells, this._menuCells, isDynamicMenuUpdateActiveBoolean, new _MenuManagerCompletionListener()
```

Comment:
When the MenuConfiguration is being updated, it updates the `_menuConfiguration` property before the operation, and the `_currentMenuConfiguration` property after the operation is successful.
Here we're building the MenuReplaceOperation with `_menuConfiguration`, so it could be possible that the configuration update fails, but this operation would still be using the configuration from the failed update instead of the most recently successful configuration that's stored in `_currentMenuConfiguration`.
Wouldn't it make sense to use the configuration that was successful?